docs/basic-features/data-fetching.md

```diff
@@ -584,6 +584,56 @@ export const getStaticPaths: GetStaticPaths = async () => {
 }
 ```
 
+If you want to get inferred typings for your context passed to `getStaticProps`, you can use `InferGetStaticPathsType<typeof getStaticPaths>`, like this:
+
+```tsx
```

Comment:
Nit: these docs are already _really_ long. Maybe we bite the bullet and break this out into an API section if you need all the options.